modules/Bio/EnsEMBL/Compara/RunnableDB/GeneTrees/StatsReport.pm

```diff
@@ -0,0 +1,94 @@
+=head1 LICENSE
+
+See the NOTICE file distributed with this work for additional information
+regarding copyright ownership.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=head1 NAME
+
+Bio::EnsEMBL::Compara::RunnableDB::GeneTrees::StatsReport
+
+=head1 DESCRIPTION
+
+Wraps gene_tree_stats.pl script and stores its output in the shared gene tree stats folder in three
+separate files, one per table generated. Each TSV file has the method type of the given MLSS ID as
+prefix.
+
+=cut
+
+package Bio::EnsEMBL::Compara::RunnableDB::GeneTrees::StatsReport;
+
+use strict;
+use warnings;
+
+use File::Path qw/make_path/;
+
+use Bio::EnsEMBL::Compara::Utils::CoreDBAdaptor;
+
+use base ('Bio::EnsEMBL::Compara::RunnableDB::BaseRunnable');
+
+
+sub param_defaults {
+    return {
+        'collection' => 'default',
+    }
+}
+
+
+sub fetch_input {
+    my $self = shift @_;
+
+    my $mlss_id = $self->param_required('mlss_id');
+    my $mlss = $self->compara_dba->get_MethodLinkSpeciesSetAdaptor->fetch_by_dbID($mlss_id);
+    $self->param('method_name', lc $mlss->method->type);
+
+    make_path($self->param_required('gene_tree_stats_shared_dir'));
```

Comment:
This should be in the `pipeline_create_commands` - I am assuming that you did not put it in using the `pipeline_create_commands_rm_mkdir` because of the `rm` part of `...rm_mkdir`. However, worry not, you can skip the `rm`: https://github.com/Ensembl/ensembl-compara/blob/bbc39e549207d9e8c09d6100744770afe401d75f/modules/Bio/EnsEMBL/Compara/PipeConfig/ComparaGeneric_conf.pm#L270

Reply:
But also, while you are there, could you change the description in the Arg to "if true" instead of "if false"? - you'll see what I mean :sweat_smile:  - my bad.